server/webSocketServer/userHolder.js

```diff
@@ -0,0 +1,40 @@
+const WebSocket = require('ws');
+const uniqid = require('uniqid');
+
+const userHolder = () => {
+    let users = [];
+
+    const addUser = connection => {
+        const id = uniqid();
+        users.push({
+            id,
+            connection,
+            name: id
+        })
+        return id;
+    };
+
+    const removeUser = id => {
+        users = users.filter(user => user.id !== id);
```

Comment:
think of using slice as recommended in react tutorial https://reactjs.org/tutorial/tutorial.html (Why Immutability Is Important)

Reply:
It's the server side BTW, and yeah in not so thrilled with this part also, but TBH this is a temp solution, this should be where i write my db querys and not saving to a local var.
also slice will not change users also, i can use splice, that actually change the array its called on but idk if this really matters. what do you think?